src/consensus/block_producer.rs

```diff
@@ -412,7 +389,7 @@ where
         let sleep_duration = duration_left.saturating_sub(Instant::now() - start_time);
         let res = tokio::select! {
             () = tokio::time::sleep(sleep_duration) => {
-                break Duration::ZERO;
+                break duration_left;
```

Comment:
I don't think I understand this change.  Didn't we use up all the time in producing the slice here?  So shouldn't we be returning 0?